dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+| authorized | abort | Payment was voided |
+| needs_init_data | abort | Payment aborted before initial data was verified |
+| none | abort | Payment aborted before initialization |
+| none | ready_for_settlement | All required tasks were performed and the payment is ready for settlement. It may happen when no payer data is required |
+| none | authorized |  All required tasks were performed and the funds are ready to be locked. It may happen when no payer data is required |
+| ready_for_settlement | abort | One party aborted the payment even though the other party already notified it is ready to settle the payment. For example, it may happen if the sender does not approve merchant data provided by the receiver |
+|
+
+### AbortCodeEnum Extension
+
+Several P2M scenarios require adding the following values to the AbortCode Enum:
+* `void` - This abort code should be used for voiding a payment that was previously authorized. For example, the merchant voided the payment because the purchased products are out of stock. It is expected that the sender VASP will respond with void code as a sign of agreement
+* `risk_checks_failed` - This abort code should be used when the risk checks performed based on the payer information failed
+
+For completeness purposes, the full list of values of the abort code enum is:
+* `rejected`
+* `void`
+* `risk_checks_failed`
+
+
+## PaymentActionObject Extension
+
+The Auth / Capture payment requires the following additions to the [PaymentActionObject](https://dip.diem.com/dip-1/#paymentactionobject):
+
+* A new field called `valid_until` added to the PaymentActionObject. This is a unix timestamp indicating until when the payment can be authorized. Once this time has been reached, the payment cannot be authorized and funds cannot be locked
+* The action enum now includes "auth" and "capture" as allowed values
+
+For completeness purposes, the PaymentActionObject now becomes:
+
+|Field|Type|Required?|Description|
+|-----|----|---------|-----------|
+| amount | uint | Y | Amount of the transfer.  Base units are the same as for on-chain transactions for this currency.  For example, if DiemUSD is represented on-chain where "1" equals 1e-6 dollars, then "1" equals the same amount here.  For any currency, the on-chain mapping must be used for amounts. |
+| currency | enum | Y | One of the supported on-chain currency types, e.g. XUS|
+| action | enum | Y | Populated in the request.  This value indicates the requested action to perform. For immediate capture and P2P transfer, "charge" is used.  For auth and capture, "auth" and "capture" are now available. "capture" can only be performed after a valid "auth" |
+| valid_until | uint | N | Unix timestamp indicating until when the payment can be authorized. Once this time has been reached, the payment cannot be authorized and funds cannot be locked. |
+| timestamp | uint | Y | Unix timestamp indicating the time that the payment Command was created.
+|
```

Comment:
these leave awkward | dangling in the text, please delete

Reply:
The way the comments are shown in github it is not clear which part of the text you want me to delete.